integration/test_servers_of_happiness.py

```diff
@@ -43,7 +43,7 @@ def test_upload_immutable(reactor, temp_dir, introducer_furl, flog_gatherer, sto
         yield proto.done
         assert False, "should raise exception"
     except Exception as e:
-        assert isinstance(e, ProcessTerminated)
+        assert "UploadUnhappinessError" in str(e)
```

Comment:
Since this is just running `tahoe ....` we're depending on the `ProcessProtocol` to raise an error when the process exits non-zero. Which it does, and includes all the output .. the "actual" exception that causes the error is of course gone at this point.

Reply:
Refactored to use its own exception (as per other comment)